conda_build/skeletons/pypi.py

```diff
@@ -254,30 +260,26 @@ def _formating_value(attribute_name, attribute_value):
 
 
 def skeletonize(
-    packages,
-    output_dir=".",
-    version=None,
-    recursive=False,
-    all_urls=False,
-    pypi_url="https://pypi.io/pypi/",
-    noprompt=True,
-    version_compare=False,
-    python_version=None,
-    manual_url=False,
-    all_extras=False,
-    noarch_python=False,
-    config=None,
-    setup_options=None,
-    extra_specs=[],
-    pin_numpy=False,
-):
+    packages: list[str],
+    output_dir: str = ".",
+    version: str | None = None,
+    recursive: bool = False,
+    all_urls: bool = False,
+    pypi_url: str = "https://pypi.io/pypi/",
+    noprompt: bool = True,
+    version_compare: bool = False,
+    python_version: str | None = None,
+    manual_url: bool = False,
+    all_extras: bool = False,
+    noarch_python: bool = False,
+    config: Config | None = None,
+    setup_options: str | Iterable[str] | None = None,
+    extra_specs: list[str] = [],
```

Comment:
Defaulting to a list in Python is not advised. There is a [pylint](https://pylint.pycqa.org/en/latest/user_guide/messages/warning/dangerous-default-value.html) rule if we would like to check for this.

https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m

Reply:
I completely agree when we seek to modify the argument, in this case `extra_specs` is not modified, it simply extends another sequence

But either way this suggests we should say the following instead:

```suggestion
    extra_specs: Iterable[str] = (),
```

Reply:
or just follow the `setup_options` pattern:

```suggestion
    extra_specs: str | Iterable[str] | None = None,
```